### PR TITLE
keychain: Drop `supports_keyring_passphrase`

### PR DESCRIPTION
**Pull request by xdustinface**
Drops the not longer needed option to disable the passphrase feature via the env variable `CHIA_PASSPHRASE_SUPPORT`.

### CODE REVIEW

File: chia/daemon/server.py
Comment (xdustinface) [inline chia/daemon/server.py:388]: I guess this needs to be addressed in the GUI?

Comment (paninaro) [reply chia/daemon/server.py:388]: Correct. I'll create a PR for the GUI to remove this dependency.

Comment (paninaro) [reply chia/daemon/server.py:388]: https://github.com/Chia-Network/chia-blockchain-gui/pull/842

File: chia/cmds/passphrase_funcs.py
Comment (paninaro) [inline chia/cmds/passphrase_funcs.py:29]: PASSPHRASE_CLI_OPTION_NAMES can be removed as well

**Comment by paninaro** [comment]
Thanks for doing this. Looks good to me with the removal of `PASSPHRASE_CLI_OPTION_NAMES`

**Comment by xdustinface** [comment]
@paninaro Dropped `PASSPHRASE_CLI_OPTION_NAMES`! But i didn't yet update the GUI because it looks like the NFT stuff isn't in `main` for `chia-blockchain-gui` but your PR was into `main`. I guess it breaks the GUI if we now merge this without your PR so we should wait until NFT gets merged into `main`?

**Comment by paninaro** [comment]
> @paninaro Dropped `PASSPHRASE_CLI_OPTION_NAMES`! But i didn't yet update the GUI because it looks like the NFT stuff isn't in `main` for `chia-blockchain-gui` but your PR was into `main`. I guess it breaks the GUI if we now merge this without your PR so we should wait until NFT gets merged into `main`?

We'll update main on chia-blockchain-gui soon. It should be fine to check this change in now, it just means that the passphrase options won't show up in the GUI settings until we merge and update the submodule pin.

**Review by paninaro — APPROVED** [review]
lgtm

**Review by emlowe — APPROVED** [review]

**Pull request merged**
